src/sql/sqlops.d.ts

```diff
@@ -108,6 +108,68 @@ declare module 'sqlops' {
 		}
 	}
 
+	/**
+	 * Namespace for interacting with Object Explorer
+	*/
+	export namespace objectexplorer {
+		/**
+		 * Get an Object Explorer node corresponding to the given connection and path. If no path
+		 * is given, it returns the top-level node for the given connection. If there is no node at
+		 * the given path, it returns undefined.
+		 * @param {string} connectionId The id of the connection that the node exists on
+		 * @param {string?} nodePath The path of the node to get
+		 * @returns {ObjectExplorerNode} The node corresponding to the given connection and path,
+		 * or undefined if no such node exists.
+		*/
+		export function getNode(connectionId: string, nodePath?: string): Thenable<ObjectExplorerNode>;
+
+		/**
+		 * Get all active Object Explorer connection nodes
+		 * @returns {ObjectExplorerNode[]} The Object Explorer nodes for each saved connection
+		*/
+		export function getActiveConnections(): Thenable<ObjectExplorerNode[]>;
```

Comment:
>getActiveConnections [](start = 18, length = 20)

getActiveConnectionNodes?

Reply:
also the result should be map of the connection id and nodes right? 


---
In reply to: [170717086](https://github.com/Microsoft/sqlopsstudio/pull/783#discussion_r170717086) [](ancestors = 170717086)